flybywiresim/aircraft
# feat(GSX): GSX Integration for Chocks and Cones

Fixes #8402

## Summary of Changes

- Enabling Fuel/Payload Sync does not disable native Chocks & Cones anymore
- Disabled GSX Cones for Aircraft Profile
- Changed parkingbrakestest in Aircraft Profile to include Chocks (Ground-Equipment)
- Changed Logic for A32NX_GND_EQP_IS_VISIBLE to also include GSX Push-Back State (removing Chocks&Cones when Push-Back becomes active)

## Screenshots (if necessary)
N/A

## References

## Additional context
I know that could be considered a GSX Automation, and that is right. But for me as long-standing GSX User a good GSX Integration includes (automatic) coordination between of Aircraft-Systems (e.g. Chocks) and GSX-Service (Push-Back). So it is kinda both 😄 
E.g. on the Fenix A320 and iniBuilds A300 both allow GSX Services to be called when their own Chocks are placed.

On Top of that, it made never Sense to me why GSX Payload/Fuel Sync disables both Chocks & Cones. One is W&B the other Ground-Equipment. So they are now decoupled 😉 

Discord username (if different from GitHub): Fragtality

## Testing instructions

1. Load into the Plane in Cold & Dark State
2. Open GSX Menu and go to "Customize Airplane" => GSX should indicate "Developer-provided" under "Aircraft data used by GSX". Else: Reset => Save => Restart Couatl
3. Open Settings in the EFB and reenable Chocks & Cones (Sim Options) while GSX Payload & Fuel Sync are active (3rd Party Options) => Should be possible. Chocks and Cones will appear.
4. Release Parking Brake and request any GSX Service (except Push-Back) => GSX should allow the Service to be called
5. When ready for Departure, Call GSX Push-Back => Chocks & Cones should be removed when GSX starts with it's Animations



## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
